### ISSUE TITLE
[P0] Scheduler state continuity on resume — LR jump 190x breaks optimization

### ISSUE THREAD

**Issue by fritzprix** [fritzprix/systemone-lite#8]
# Issue: Scheduler State Continuity on Resume

## Problem

`--resume` does **not** guarantee optimizer/scheduler continuity.

In run `spatial-v2-s1-cloze` (52,124 steps), the learning rate jumped **190×** at resume:

| Time | LR |
|---|---|
| Pre-resume (step ~20k) | `7.78e-8` |
| Resume (step ~20,003) | `1.48e-5` |

Root cause: first run used `max_steps=20000` for scheduler, resume used `max_steps=60200`, causing the scheduler to recompute LR at the same global step with a different horizon.

## Impact

- Training loss worsened after resume (0.875 → 1.30)
- Validation accuracy plateaued at 54% (40k spike of 56% unreproducible)
- Run deemed inconclusive — optimization trajectory fractured

### 1. Save full optimizer/scheduler state on checkpoint

```python
# On save:
ckpt = {
    "model_state": model.state_dict(),
    "optimizer_state": optimizer.state_dict(),
    "scheduler_state": scheduler.state_dict(),  # ← critical
    "amp_scaler": scaler.state_dict() if amp else None,
    "rng_states": {
        "python": random.getstate(),
        "numpy": np.random.get_state(),
        "torch": torch.random.get_rng_state(),
        "cuda": torch.cuda.get_rng_state_all() if cuda else None,
    },
    "sampler_position": sampler.get_position() if sampler else None,
    "last_lr": optimizer.param_groups[0]["lr"],
    "global_step": step,
}
```

### 2. Verify continuity on resume

```python
# On resume:
saved_lr = ckpt["last_lr"]
resumed_lr = optimizer.param_groups[0]["lr"]
lr_jump = abs(resumed_lr - saved_lr) / max(saved_lr, 1e-12)

if lr_jump > 0.05:
    raise RuntimeError(
        f"Unsafe LR discontinuity: {saved_lr} -> {resumed_lr} "
        f"(jump={lr_jump:.1%}). Resume aborted."
    )
```

### 3. Fix total horizon policy

If planning 60,200 total steps:
- Use `max_steps=60200` from the **first** run
- Treat 20k as a checkpoint boundary, NOT a scheduler end point
- If total steps must change mid-training, treat it as a **new continuation experiment** (new W&B run), not a resume of the same run

## Acceptance Criteria

- [ ] `optimizer.state_dict()` saved in every checkpoint
- [ ] `scheduler.state_dict()` saved in every checkpoint
- [ ] LR continuity check on resume with 5% tolerance
- [ ] `max_steps` immutable within a single W&B run
- [ ] Test: resume with different `max_steps` fails with clear error message

**Comment by fritzprix** [comment]
## Fix landed (local)

`scripts/chess_finetune.py` now enforces GitHub #8 policy:

1. **`max_steps` immutable on resume** — `configs_compatible` requires exact config match (no raising horizon mid-run).
2. **`last_lr` saved** in `train_state.pt` + `checkpoint_meta.json`.
3. **LR continuity check** (5% relative) after loading optimizer/scheduler — aborts on jump.
4. **`total_steps` horizon check** — aborts if checkpoint horizon ≠ current schedule.

Tests: `tests/test_finetune_resume_guards.py`.

### Contaminated run handling
- Clean pre-resume 20k weights preserved at `checkpoints/systemone-spatial-v2-s1-pre-resume/` (same as Hub `dwidlee/systemone-lite-0.5b`).
- `step-50000..52000` under `systemone-spatial-v2-s1` marked contaminated (`CONTAMINATED_RESUME.json`) — do not publish.

Next: LR pilot (#12) from the clean 20k, as a **new** W&B run (not resume of `bs62qs8x`).
